### PR TITLE
QGA-5 create record model

### PR DESCRIPTION
**Pull request by Bekhzod96**
Model has been created

### CODE REVIEW

File: models/recordsModel.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    answers: [{ type: Number, requred: true }]
+  },
+  { timestamps: true }
+);
+
+const RecordModel = mongoose.model("Record", recordSchema);
+
+const save = async model => new RecordModel(model).save();
+const getAllRecords = async () => RecordModel.find();
+const getRecordById = async recordId => RecordModel.findOne(recordId);
```
Comment (Gruseckis) [inline models/recordsModel.js:14]: This should be findById not find one.

File: models/recordsModel.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const updateById = async (recordId, update) =>
+  RecordModel.findOneAndUpdate(recordId, update, { new: true });
```
Comment (Gruseckis) [inline models/recordsModel.js:17]: This should be findByIdAndUpdate

File: models/recordsModel.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+const deleteRecordById = async recordId =>
+  RecordModel.findOneAndDelete(recordId);
```
Comment (Gruseckis) [inline models/recordsModel.js:20]: This should be findByIdAndDelete

File: models/recordsModel.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+// What acctual data we need to handle throuw
+const getRecordsFromIdArray = async recordId =>
+  RefordModel.find({
+    recordID: { $in: [mongoose.Types.ObjectId(recordId.map())] }
```
Comment (Gruseckis) [inline models/recordsModel.js:25]: This wont work for few reasons.

- You need to specify what field you want to make query agains. I think it should be ID
- You are using map function but I think you do not know how it work. I would suggest creating array before and then pass it into function. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

File: models/recordsModel.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+export { save, getRecordById, updateById, deleteRecordById, getAllRecords };
```
Comment (Gruseckis) [inline models/recordsModel.js:28]: You need to export Schema and model as well as all functions that might be used outside of this model. And make sure they are correct names

File: routes/recordsRoute.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import express from "express";
```
Comment (Gruseckis) [inline routes/recordsRoute.js:1]: This should not be a part of this Pull request

**Review by Gruseckis — CHANGES_REQUESTED** [review]

**Comment by Gruseckis** [comment]
You need to change target to development

**Comment by Gruseckis** [comment]
Duplicate

File: models/recordsModel.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import mongoose from "mongoose";
+const recordSchema = new mongoose.Schema(
+  {
+    questionId: { type: String, required: true, unique: false },
+    answers: [{ type: Number, requred: true }]
```
Comment (sadinovsr) [inline models/recordsModel.js:5]: There is a typo for required.

**Pull request closed without merging**